### PR TITLE
#[UIC-1657] provide a single API to create slices and add it to dashboard

### PR DESCRIPTION
**Pull request by rasmi-ranjan-guavus**
UIC-1657

### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This will create new dashboard and add multiple charts in the newly created dashboard

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@jitendra-kumawat

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -290,7 +292,7 @@ def create(self):
         except Exception as e:
             logging.exception(e)
             return json_error_response(e)
- 
+
         return json_success(json.dumps({'database_id': db_model.id}))
 
     def pre_add(self, db):
```
Comment (jitendra-kumawat) [inline superset/views/core.py]: remove unwanted lines added

File: superset/views/core.py [hunk 2]
```diff
@@ -857,6 +864,26 @@ def addtodashboard(self):
...
+                _slice = update_slice_metadata(_slice)
+
+                slice_param_data = {
+                  'action':	'saveas',
+                  'slice_name':	_slice['slice_name'],
+                  'add_to_dash':	'existing',
+                  'save_to_dashboard_id':	json.loads(new_dashboard.content)['dashboard_id'],
+                  'goto_dash':	'false',
+                }
+
```
Comment (jitendra-kumawat) [inline superset/views/core.py]: remove white spaces  in property and value

File: superset/views/core.py [hunk 3]
```diff
@@ -2013,10 +2040,10 @@ def user_slices(self, user_id=None):
...
         )
-        
-        
+
+
         payload = []
-        
+
         for o in qry.all():
             publish_columns = []
```
Comment (jitendra-kumawat) [inline superset/views/core.py]: remove unwanted lines

File: superset/views/dashboard.py [hunk 4]
```diff
@@ -40,4 +43,20 @@ def new(self):
...
+            slug=request.form.get('slug'),
+        )
+        db.session.add(new_dashboard)
+        db.session.commit()
+        logging.info('new dashboard created with name = ' + request.form.get('dashboard_title'))
+
+        return Response(
+            json.dumps({'dashboard_id': new_dashboard.id}),
+            status=200)
+
```
Comment (jitendra-kumawat) [inline superset/views/dashboard.py]: use json_sucess instead manually creating Response

File: superset/views/default_slice_metadata.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                if prop not in metric:
+                  metric[prop] = SIMPLE_ADHOC_FILTER[prop]
+                  if prop == 'column':
+                    for column_prop in DEFAULT_COLUMN:
+                      if column_prop in metric[prop]:
+                        metric[prop][column_prop] = DEFAULT_COLUMN[column_prop]
+
+    return slice
+
+
```
Comment (jitendra-kumawat) [inline superset/views/default_slice_metadata.py]: unit test cases for validate logic is missing, we need to define default schema for Line and Bar chart also

File: superset/views/core.py (hunk 1 above)
Comment (rasmi-ranjan-guavus) [reply superset/views/core.py]: done

File: superset/views/core.py (hunk 2 above)
Comment (rasmi-ranjan-guavus) [reply superset/views/core.py]: done

File: superset/views/dashboard.py (hunk 4 above)
Comment (rasmi-ranjan-guavus) [reply superset/views/dashboard.py]: done

File: superset/views/core.py (hunk 3 above)
Comment (rasmi-ranjan-guavus) [reply superset/views/core.py]: done

**Pull request merged**